satellite/gracefulexit/endpoint.go

```diff
@@ -497,6 +497,16 @@ func (endpoint *Endpoint) handleSucceeded(ctx context.Context, pending *pendingM
 		return Error.New("Transfer path cannot be nil")
 	}
 
+	defer func() {
```

Comment:
Try to avoid defers for logic control since they make the code much harder to read.